### ISSUE TITLE
Make changes in naming strategy in MockMiddleware

### ISSUE THREAD

**Issue by adek23** [csarrazi/CsaGuzzleBundle#187]
First thing, why **$namingStrategies** variable is private in **MockStorageAdapter**?
This way, its hard to extend this class.

Next thing, why **AbstractNamingStrategy** in method **getFingerprint** doesn't use request **body**?

And the last thing - naming strategies should be injected, not instantized.

**I can make PR if you agree with my suggestions.**

**Comment by csarrazi** [comment]
Hi @adek23,

1. The naming strategies were not exposed for one simple reason. They were exposed as a fix in a patch version, which means that it was not meant to bring a new feature to the bundle. Also, they were not meant to be exposed at the time. As you may notice that there is currently no semantic configuration for setting a naming strategy.
2. Regarding the fingerprint, I have nothing against including the body.
3. See point 1. At the time, it was pointless to inject them as the logic was directly implemented in each adapter. This was actually the first step to a general refactoring of the adapters, in order to merge most of the MockStorageAdapter's logic with the AbstractCacheAdapter. Keep in mind that at first, this logic was actually implemented directly in the storage adapter, and not separately.

**TL;DR**: I have absolutely nothing against exposing the naming strategies. I am completely fine with someone making a PR to improve this! 😄

**Comment by csarrazi** [comment]
Ping @adek23. Still intend to make a PR? :)

**Comment by csarrazi** [comment]
@adek23 I'm closing the issue in this project, but I have opened a new issue in csarrazi/guzzle-cache-middleware.

Do you still intend to make a PR?
